src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -33,12 +44,219 @@ public static class DefaultAssemblyPartDiscoveryProvider
+            Assembly viewsAssembly = null;
+            if (AssemblyResolver(filePath))
+            {
+                try
+                {
+                    viewsAssembly = AssemblyLoader(filePath);
+                }
+                catch (FileLoadException)
+                {
+                    // Don't throw if assembly cannot be loaded. This can happen if the file is not a managed assembly.
+                }
+            }
+
+            return viewsAssembly;
+        }
+
+        private static Assembly LoadFromConvention(Assembly assembly)
+        {
+            if (assembly.IsDynamic || string.IsNullOrEmpty(assembly.Location))
+            {
+                return null;
+            }
+
+            for (var i = 0; i < ViewAssemblySuffixes.Count; i++)
+            {
+                var fileName = assembly.GetName().Name + ViewAssemblySuffixes[i] + ".dll";
+                var filePath = Path.Combine(Path.GetDirectoryName(assembly.Location), fileName);
+
+                var viewsAssembly = LoadAssembly(filePath);
+                if (viewsAssembly != null)
+                {
+                    return viewsAssembly;
+                }
+            }
+
+            return null;
+        }
+
+        private static void AddAdditionalReference(
+            Assembly additionalReference,
+            Assembly entryAssembly,
+            Assembly assembly,
+            SortedSet<Assembly> additionalReferences,
+            SortedSet<Assembly> entryAssemblyAdditionalReferences)
+        {
+            if (additionalReference == null ||
+                additionalReferences.Contains(additionalReference) ||
+                entryAssemblyAdditionalReferences.Contains(additionalReference))
+            {
+                return;
+            }
+
+            if (assembly.Equals(entryAssembly))
+            {
+                entryAssemblyAdditionalReferences.Add(additionalReference);
+            }
+            else
+            {
+                additionalReferences.Add(additionalReference);
+            }
+        }
+
+        internal class AdditionalReferencesPair
+        {
+            public SortedSet<Assembly> AdditionalAssemblies { get; set; }
+            public SortedSet<Assembly> EntryAssemblyAdditionalAssemblies { get; set; }
+
+            public void Deconstruct(
+                out SortedSet<Assembly> additionalAssemblies,
+                out SortedSet<Assembly> entryAssemblyAdditionalAssemblies)
+            {
+                additionalAssemblies = AdditionalAssemblies;
+                entryAssemblyAdditionalAssemblies = EntryAssemblyAdditionalAssemblies;
+            }
+        }
+
+        internal class FullNameAssemblyComparer : IComparer<Assembly>
+        {
+            public static IComparer<Assembly> Instance { get; } = new FullNameAssemblyComparer();
+
+            public int Compare(Assembly x, Assembly y) =>
+                string.Compare(x?.FullName, y?.FullName, StringComparison.Ordinal);
```

Comment:
A few thoughts about this:
- I'm not sure how you could end up with a version conflict
- If you did, those would be treated as separate assemblies
- I'm not sure if that's bad

Just a comment. I don't have any suggestions

Reply:
@rynowak I think if you have a version conflict it will fail at a later point in time if you try to reference one of the types, won't that be the case? (At least when recompiling).

I don't think that it's likely you end up with a version conflict here either, what would be the downside of just using the assembly name? (I'm not an expert in this topic)